src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
```

Comment:
I noticed you are using a let variable called sum to accumulate values, and this approach works well. However, the reduce method could be a more concise and functional alternative for performing this summation. It not only simplifies the code but also makes the intention to sum values clearer. I recommend considering the implementation of reduce to make the code more efficient.